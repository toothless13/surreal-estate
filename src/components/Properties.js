import React from "react";
import PropertyCard from "./PropertyCard";
import "../styles/properties.css";

const Properties = () => {
  return (
    <>
      <div className="property-page-heading">Properties Page</div>
      <div className="property-cards-container">
        <PropertyCard
          title="title"
          type="Flat"
          bedrooms="3"
          bathrooms="1"
          price="12345"
          city="city"
          email="test@email.com"
        />
      </div>
    </>
  );
};

export default Properties;
