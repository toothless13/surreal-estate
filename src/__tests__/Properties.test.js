import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Properties from "../components/Properties";

describe("Properties", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Properties />
      </MemoryRouter>,
    );

    const headingElement = screen.getByText(/Properties Page/i);

    expect(asFragment()).toMatchSnapshot();
    expect(headingElement).toBeTruthy();
  });
});
