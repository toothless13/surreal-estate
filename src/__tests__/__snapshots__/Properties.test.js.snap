// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-page-heading"
  >
    Properties Page
  </div>
  <div
    class="property-cards-container"
  >
    <div
      class="property-card"
    >
      <p
        class="property-card-title"
      >
        title
      </p>
      <p
        class="property-card-type"
      >
        Flat
      </p>
      <p
        class="property-card-bedrooms"
        data-testid="property-bedrooms"
      >
        3
      </p>
      <p
        class="property-card-bathrooms"
        data-testid="property-bathrooms"
      >
        1
      </p>
      <p
        class="property-card-price"
      >
        12345
      </p>
      <p
        class="property-card-city"
      >
        city
      </p>
      <a
        class="property-card-email"
        href="mailto:test@email.com"
      >
        Email
      </a>
    </div>
  </div>
</DocumentFragment>
`;
